modules/cli/src/main/scala/scala/cli/exportCmd/MavenProjectDescription.scala

```diff
@@ -0,0 +1,236 @@
+            if (scope == Scope.Test) Some("test")
+            else if (isCompileOnly) Some("provided")
+            else None
+
+          MavenLibraryDependency(org, artNameWithPrefix, ver, scope0)
+        }
+
+        val scalaDep = if (!ProjectDescriptor.isPureJavaProject(options, sources)) {
+          // add scala dependency
+          // todo: get scala version from directive
+          val scalaDep = if scalaV.startsWith("3") then "scala3-library_3" else "scala-library"
+          List(MavenLibraryDependency("org.scala-lang", scalaDep, scalaV))
+        }
+        else Nil
+
+        providedDeps ++ scalaDep
+      }
+
+      toDependencies(options.classPathOptions.extraDependencies, true)
+    }
+
+    MavenProject(
+      dependencies = depSettings
+    )
+  }
+
+  private def getScalaVersion(options: BuildOptions): String =
+    options.scalaOptions.scalaVersion
+      .flatMap(_.versionOpt) // FIXME If versionOpt is empty, the project is pure Java
+      .getOrElse(ScalaCli.getDefaultScalaVersion)
+
+  private def plugins(
+    options: BuildOptions,
+    scope: Scope,
+    jdkVersion: String,
+    sourcesMain: Sources
+  ): MavenProject = {
+
+    // todo: use this method from mill and sbt projects as well
+    val pureJava = ProjectDescriptor.isPureJavaProject(options, sourcesMain)
+
+    val javacOptions = javacOptionsSettings(options)
+
+    val mavenJavaPlugin = buildJavaCompilerPlugin(javacOptions, jdkVersion)
+    val scalaPlugin     = buildScalaPlugin(javacOptions, jdkVersion, getScalaVersion(options))
+
+    val reqdPlugins = if (pureJava) Seq(mavenJavaPlugin) else Seq(scalaPlugin)
+
+    MavenProject(
+      plugins = reqdPlugins
+    )
+  }
+
+  private def buildScalaPlugin(
+    javacOptions: Seq[String],
+    jdkVersion: String,
+    scalaVersion: String
+  ): MavenPlugin = {
+
+    val compileMode = buildNode("recompileMode", "incremental")
+    val scalaVersionNode =
+      buildNode("scalaVersion", scalaVersion) // todo: set this value in properties
+    val javacOptionsElem = {
+      val opts = javacOptions.map { opt =>
+        buildNode("javacArg", opt)
+      }
+      <javacArgs>
+        {opts}
+      </javacArgs>
+    }
+
+    val configurationElements = Seq(compileMode, /*scalaVersionNone,*/ javacOptionsElem)
+
+    MavenPlugin(
+      "net.alchim31.maven",
+      "scala-maven-plugin",
+      "4.9.1",
+      jdkVersion,
+      configurationElements
+    )
+  }
+
+  private def buildJavaCompilerPlugin(
+    javacOptions: Seq[String],
+    jdkVersion: String
+  ): MavenPlugin = {
+    val javacOptionsElem = {
+      val opts = javacOptions.map { opt =>
+        buildNode("arg", opt)
+      }
+      <compilerArgs>
+        {opts}
+      </compilerArgs>
+    }
+
+    val sourceArg = buildNode("source", jdkVersion)
+    val targetArg = buildNode("target", jdkVersion)
+
+    MavenPlugin(
+      "org.apache.maven.plugins",
+      "maven-compiler-plugin",
+      "3.8.1",
```

Comment:
similarly, this version should be configurable and the defaults passed via `Constants`